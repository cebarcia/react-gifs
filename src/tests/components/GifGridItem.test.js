import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import React from "react";
import { GifGridItem } from "../../components/GifGridItem";

describe("Test GifGridItem ", () => {
  const title = "A title";
  const url = "https://localhost/something.jpg";

  let wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("Test if GifGridItem is showing correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("the title must have a paragraph", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("must have an image with the same way of url and an alt iin the props", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("must have animate__fadeIn", () => {
    const div = wrapper.find("div");

    expect(div.props("className").className).toContain("animate__faceIn");
  });
});
