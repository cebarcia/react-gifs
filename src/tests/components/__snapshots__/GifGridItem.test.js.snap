// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem  Test if GifGridItem is showing correctly 1`] = `
<div
  className="card animate__animated animate__bounce animate__faceIn"
>
  <img
    alt="A title"
    src="https://localhost/something.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
